src/main/java/pl/edu/pw/mini/ingreedio/api/auth/service/AuthService.java

```diff
@@ -1,98 +1,102 @@
 import org.springframework.security.authentication.AuthenticationManager;
 import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
 import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
 import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
-import pl.edu.pw.mini.ingreedio.api.auth.dto.AuthRequestDto;
-import pl.edu.pw.mini.ingreedio.api.auth.dto.JwtResponseDto;
-import pl.edu.pw.mini.ingreedio.api.auth.dto.RefreshTokenRequestDto;
-import pl.edu.pw.mini.ingreedio.api.auth.dto.RegisterRequestDto;
+import org.zalando.problem.Problem;
+import org.zalando.problem.Status;
+import org.zalando.problem.ThrowableProblem;
+import pl.edu.pw.mini.ingreedio.api.auth.exception.UserAlreadyExistsException;
 import pl.edu.pw.mini.ingreedio.api.auth.model.AuthInfo;
 import pl.edu.pw.mini.ingreedio.api.auth.model.RefreshToken;
-import pl.edu.pw.mini.ingreedio.api.auth.repository.AuthRepository;
+import pl.edu.pw.mini.ingreedio.api.auth.model.Role;
+import pl.edu.pw.mini.ingreedio.api.auth.repository.AuthInfoRepository;
+import pl.edu.pw.mini.ingreedio.api.auth.security.JwtAuthTokens;
 import pl.edu.pw.mini.ingreedio.api.user.model.User;
-import pl.edu.pw.mini.ingreedio.api.user.repository.UserRepository;
 
 @Service
 @RequiredArgsConstructor
 public class AuthService {
-    private final AuthRepository authRepository;
-    private final UserRepository userRepository;
-
     private final AuthenticationManager authenticationManager;
-    private final JwtService jwtService;
     private final PasswordEncoder passwordEncoder;
+
+    private final AuthInfoRepository authInfoRepository;
+
+    private final JwtService jwtService;
     private final RefreshTokenService refreshTokenService;
     private final JwtClaimsService jwtClaimsService;
     private final RoleService roleService;
 
     @Transactional
-    public User register(RegisterRequestDto request) {
-        User user = User.builder()
-            .displayName(request.displayName())
-            .email(request.email())
-            .build();
-
+    public AuthInfo register(String username, String password, User user)
+        throws UserAlreadyExistsException {
         AuthInfo authInfo = AuthInfo.builder()
-            .username(request.username())
-            .password(passwordEncoder.encode(request.password()))
+            .username(username)
+            .password(passwordEncoder.encode(password))
             .user(user)
             .roles(roleService.getDefaultUserRoles())
             .build();
 
-        userRepository.save(user);
-        authRepository.save(authInfo);
+        try {
```

Comment:
I think that we can create separate service `AuthInfoManagerService` (or smth like that) that has `addUser` function which throws `UserAlreadyExistsException` and other necessary functions for user db management. Then in this service, the `AuthInfoRepository` would be replaced with the `AuthInfoManagerService` and there would be only one place where the repository is attached. 

On the other hand we would use that these function only here so maybe this layering is useless. Do whatever you want with this comment 😊

Reply:
I considered that, per your request in https://github.com/Java-DZGS/ingreedio-api/issues/78#issuecomment-2128184402 but I decided against it at the moment of this commit, but I think you convinced me to implement such a solution 🥰